stream.go

```diff
@@ -56,6 +56,7 @@ var (
 type SubscriptionError struct {
 	Code    int
 	Message string
+	Headers http.Header
```

Comment:
We need to expose the headers on subscription failures so we can look for the `X-LD-FD-Fallback` header with FDv2.